### PR TITLE
URL encode postgres connection string

### PR DESCRIPTION
**Pull request by jgpruitt**
Postgres connection strings should be valid URLs. User names, passwords, database names,
application names, etc. can all contain characters that must be escaped in a URL. This
change uses the url package to construct the connection string and thus ensures that
it is properly escaped.

Addresses issue #727

### CODE REVIEW

**Review by cevian — APPROVED** [review]
LGTM

**Review by JamesGuthrie — APPROVED** [review]

File: pkg/pgclient/config.go [hunk 1]
```diff
@@ -119,15 +121,18 @@ func (cfg Config) validateConnectionSettings() error {
...
-			cfg.Host,
-			cfg.Port,
-			cfg.Database,
-			cfg.AppName,
-			cfg.SslMode,
-			int(cfg.DbConnectionTimeout.Seconds()))
+		v := url.Values{}
+		v.Set("application_name", cfg.AppName)
+		v.Set("sslmode", cfg.SslMode)
+		v.Set("connect_timeout", strconv.Itoa(int(cfg.DbConnectionTimeout.Seconds())))
```
Comment (antekresic) [inline pkg/pgclient/config.go:127]:
```suggestion
		v.Set("connect_timeout", fmt.Sprintf("%.f", cfg.DbConnectionTimeout.Seconds()))
```

**Review by antekresic — COMMENTED** [review]
Minor change to avoid double conversion.

**Pull request merged**